doc/man3/OSSL_PARAM_build.pod

```diff
@@ -0,0 +1,191 @@
+=pod
+
+=head1 NAME
+
+OSSL_PARAM_build, OSSL_PARAM_build_free,
+OSSL_PARAM_BUILD_BN, OSSL_PARAM_BUILD_double,
+OSSL_PARAM_BUILD_int, OSSL_PARAM_BUILD_int32,
+OSSL_PARAM_BUILD_int64, OSSL_PARAM_BUILD_long,
+OSSL_PARAM_BUILD_size_t, OSSL_PARAM_BUILD_uint,
+OSSL_PARAM_BUILD_uint32, OSSL_PARAM_BUILD_uint64,
+OSSL_PARAM_BUILD_ulong, OSSL_PARAM_BUILD_BN,
+OSSL_PARAM_BUILD_utf8_string,
+OSSL_PARAM_BUILD_octet_string,
+- OSSL_PARAM array helpers
+
+=head1 SYNOPSIS
+
+=for comment generic
+
+ #include <openssl/params.h>
+
+ #define OSSL_PARAM_BUILD_TYPE(key, address)
+ #define OSSL_PARAM_BUILD_utf8_string(key, address, size)
+ #define OSSL_PARAM_BUILD_octet_string(key, address, size)
+ #define OSSL_PARAM_BUILD_BN(key, address, size)
+
+ OSSL_PARAM *OSSL_PARAM_build(const char *name, ..., NULL);
+ void OSSL_PARAM_build_free(OSSL_PARAM *params);
+
+=head1 DESCRIPTION
+
+A collection of utility functions that simplify building arrays of
+OSSL_PARAM structures.  The following B<TYPE> names are supported:
+
+=over 1
+
+=item *
+
+double
+
+=item *
+
+int
+
+=item *
+
+int32 (int32_t)
+
+=item *
+
+int64 (int64_t)
+
+=item *
+
+long int (long)
+
+=item *
+
+size_t
+
+=item *
+
+uint32 (uint32_t)
+
+=item *
+
+uint64 (uint64_t)
+
+=item *
+
+unsigned int (uint)
+
+=item *
+
+unsigned long int (ulong)
+
+=back
+
+OSSL_PARAM_BUILD_TYPE() are a series of macros designed to assist calling the
+OSSL_PARAM_build() function.
+In each case a parameter record is defined with the specified B<name> and
+B<value>.
+Each of these macros expands into a series of arguments appropriate for passing
+to the OSSL_PARAM_build() function which reduces the chance of parameters
+mistakes.
+Casts are also inserted to again reduce typing mistakes.
+
+OSSL_PARAM_BUILD_utf8_string(), OSSL_PARAM_BUILD_octet_string(),
+OSSL_PARAM_BUILD_BN() are macros that provide support
+for defining UTF8 strings, OCTET strings and big numbers.
+A parameter with B<name> is defined, storage of length B<len> is allocated and
+the B<value> copied.
+If the B<value> is NULL, the allocated memory will be zeroed.
+If B<len> is zero, the length will be calculated for
+OSSL_PARAM_BUILD_utf8_string() and OSSL_PARAM_BUILD_BN().
+For OSSL_PARAM_BUILD_octet_string() the length must be specified.
+
+OSSL_PARAM_build() allocates storage for and creates an array of OSSL_PARAM
+structures and their associated values.
+The parameter list must be terminated with a NULL.
+The order of the elements in the allocated OSSL_PARAM array are undefined.
```

Comment:
IS undefined, not are.

Reply:
I was tossing this one up either way.
I'll switch to _is_.

Reply:
The prepositional phrase "of the elments" and "in the allocated OSSL param array" don't enter into the subject/verb agreement, so "The order *IS* undefined" :)